### PR TITLE
feat: Checkbox support cssvar

### PR DESCRIPTION
**Pull request by RedJue**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [X] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
- #45618 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Checkbox support cssvar       |
| 🇨🇳 Chinese |     多选框支持 css 变量    |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 78289f1</samp>

This pull request refactors the Checkbox and Checkbox.Group components to use CSS variables for theming and styling. It introduces a new `useCSSVar` hook that manages the CSS variables and a new file `components/checkbox/style/cssVar.ts` that exports it. It also improves the consistency and customizability of the component's style by using the `unit` and `calc` functions from `@ant-design/cssinjs`.

### <samp>🤖 Generated by Copilot at 78289f1</samp>

*  Replace `wrapSSR` with `wrapCSSVar` for Checkbox and Checkbox.Group components to use CSS variables instead of style tags for server-side rendering ([link](https://github.com/ant-design/ant-design/pull/45859/files?diff=unified&w=0#diff-e5138148a71ae8aea8e59e4b8934815b003ff8836bbb6ca6dc91d41aef32ef86L107-R109), [link](https://github.com/ant-design/ant-design/pull/45859/files?diff=unified&w=0#diff-e5138148a71ae8aea8e59e4b8934815b003ff8836bbb6ca6dc91d41aef32ef86L143-R145), [link](https://github.com/ant-design/ant-design/pull/45859/files?diff=unified&w=0#diff-d3e984f0984d6a44c780ab883b2f78b2df8029b1ce515d879e60879bb425fa63L112-R114), [link](https://github.com/ant-design/ant-design/pull/45859/files?diff=unified&w=0#diff-d3e984f0984d6a44c780ab883b2f78b2df8029b1ce515d879e60879bb425fa63L154-R156))
*  Create and export `useCSSVar` hook for Checkbox component in `cssVar.ts` file using `genCSSVarRegister` utility function ([link](https://github.com/ant-design/ant-design/pull/45859/files?diff=unified&w=0#diff-3dd83ffba2c18472c0e62ae7d7e3cf74750c07ec843fab598394579b9b2ce5adR1-R3))
*  Import and use `useCSSVar` hook in `Checkbox.tsx` and `Group.tsx` files to register and update CSS variables for Checkbox component and subcomponent ([link](https://github.com/ant-design/ant-design/pull/45859/files?diff=unified&w=0#diff-e5138148a71ae8aea8e59e4b8934815b003ff8836bbb6ca6dc91d41aef32ef86R14), [link](https://github.com/ant-design/ant-design/pull/45859/files?diff=unified&w=0#diff-d3e984f0984d6a44c780ab883b2f78b2df8029b1ce515d879e60879bb425fa63R10))
*  Import and use `unit` function from `@ant-design/cssinjs` library to convert numeric values to CSS unit strings in `genCheckboxStyle` and `genFocusOutline` functions ([link](https://github.com/ant-design/ant-design/pull/45859/files?diff=unified&w=0#diff-c624f3acd6eb939a50b97d0fe9f815bd0dd657eb19797790a8e7b1755ee550efR1-R2), [link](https://github.com/ant-design/ant-design/pull/45859/files?diff=unified&w=0#diff-c624f3acd6eb939a50b97d0fe9f815bd0dd657eb19797790a8e7b1755ee550efL105-R107), [link](https://github.com/ant-design/ant-design/pull/45859/files?diff=unified&w=0#diff-c624f3acd6eb939a50b97d0fe9f815bd0dd657eb19797790a8e7b1755ee550efL116-R120), [link](https://github.com/ant-design/ant-design/pull/45859/files?diff=unified&w=0#diff-c624f3acd6eb939a50b97d0fe9f815bd0dd657eb19797790a8e7b1755ee550efL198-R201), [link](https://github.com/ant-design/ant-design/pull/45859/files?diff=unified&w=0#diff-6ddd1e92016eb62920ed5a9e2a23fa555dc551e5c9a2e7c78c128919f357adaaL2-R3), [link](https://github.com/ant-design/ant-design/pull/45859/files?diff=unified&w=0#diff-6ddd1e92016eb62920ed5a9e2a23fa555dc551e5c9a2e7c78c128919f357adaaL131-R132))
*  Use `token.calc` function to perform arithmetic operations on CSS variables in `genCheckboxStyle` function and return CSS calc expression strings ([link](https://github.com/ant-design/ant-design/pull/45859/files?diff=unified&w=0#diff-c624f3acd6eb939a50b97d0fe9f815bd0dd657eb19797790a8e7b1755ee550efL116-R120), [link](https://github.com/ant-design/ant-design/pull/45859/files?diff=unified&w=0#diff-c624f3acd6eb939a50b97d0fe9f815bd0dd657eb19797790a8e7b1755ee550efL198-R201))

### CODE REVIEW

File: components/checkbox/Checkbox.tsx [hunk 1]
```diff
@@ -104,7 +105,8 @@ const InternalCheckbox: React.ForwardRefRenderFunction<CheckboxRef, CheckboxProp
   }, [restProps.value]);
 
   const prefixCls = getPrefixCls('checkbox', customizePrefixCls);
-  const [wrapSSR, hashId] = useStyle(prefixCls);
+  const [, hashId] = useStyle(prefixCls);
+  const wrapCSSVar = useCSSVar(prefixCls);
```
Comment (MadCcc) [inline components/checkbox/Checkbox.tsx:109]: 这里应该用 wrapper，prefixCls 不在最外层

Comment (MadCcc) [reply components/checkbox/Checkbox.tsx:109]: 建议用 useCSSVarCls 拿一个 rootCls 吧，和 group 可以用同一个了

Comment (RedJue) [reply components/checkbox/Checkbox.tsx:109]: 1

File: components/checkbox/Checkbox.tsx [hunk 2]
```diff
@@ -104,7 +106,9 @@ const InternalCheckbox: React.ForwardRefRenderFunction<CheckboxRef, CheckboxProp
   }, [restProps.value]);
 
   const prefixCls = getPrefixCls('checkbox', customizePrefixCls);
-  const [wrapSSR, hashId] = useStyle(prefixCls);
+  const [, hashId] = useStyle(prefixCls);
+  const rootCls = useCSSVarCls(prefixCls);
```
Comment (MadCcc) [inline components/checkbox/Checkbox.tsx:110]: rootCls 要写到组件上

Comment (RedJue) [reply components/checkbox/Checkbox.tsx:110]: 1

**Review by MadCcc — APPROVED** [review]

**Pull request merged**